applications/tari_validator_node/src/p2p/services/epoch_manager/base_layer_epoch_manager.rs

```diff
@@ -93,10 +108,31 @@ impl BaseLayerEpochManager {
         }
 
         // If the committee size is bigger than vns.len() then this function is broken.
+        let height = epoch.to_height();
         let mut base_node_client = self.base_node_client.clone();
-        let mut vns = base_node_client.get_validator_nodes(epoch.0 * 10).await?;
+        let mut vns = base_node_client.get_validator_nodes(height).await?;
         vns.sort_by(|a, b| a.shard_key.partial_cmp(&b.shard_key).unwrap());
 
+        let vn_shard_key = vns
+            .iter()
+            .find(|v| v.public_key == *self.node_identity.public_key())
+            .ok_or(EpochManagerError::UnexpectedResponse)?
+            .shard_key;
+
+        // from current_shard_key we can get the corresponding vns committee
+        let committee_vns = self.get_committee_vns_from_shard_key(epoch, vn_shard_key)?;
+        let (start_shard_id, end_shard_id) = get_committee_shard_range(&committee_vns).into_inner();
+
+        let peer_sync_service_manager = PeerSyncManagerService::new(
+            self.validator_node_config.clone(),
+            self.validator_node_client_factory.clone(),
+        );
+
+        // synchronize state with committee validator nodes
+        peer_sync_service_manager
+            .sync_peers_state(committee_vns, start_shard_id, end_shard_id, vn_shard_key)
+            .await?;
+
```

Comment:
Move this code to after the epoch is processed (line 151)

```rust
// TODO: sync should be triggered by EpochChanged event
```

Add a scope (`{ ... }`) from line 138 to line 146 so that the transaction and db is cleaned up

Reply:
Makes sense, thanks for the remarks and explanations !